tests/lib/hardware/openstack_libcloud.py

```diff
@@ -85,14 +86,19 @@ def boot(self, size, image, sshkey_name=None, additional_networks=[],
         print(self)
         print(self.libcloud_node)
 
-    def create_and_attach_floating_ip(self):
-        # TODO(jhesketh): Move cloud-specific configuration elsewhere
-        floating_ip = self.libcloud_conn.ex_create_floating_ip(
-            config.OS_EXTERNAL_NETWORK)
-
-        print("Created floating IP: ")
-        print(floating_ip)
-        self.floating_ips.append(floating_ip)
+    def get_floating_ip(self):
+        try:
+            floating_ip = self.libcloud_conn.ex_create_floating_ip(
+                config.OS_EXTERNAL_NETWORK)
+            print("Created floating IP: ")
+            print(floating_ip)
+            self.floating_ips.append(floating_ip)
+            self.floating_ips_created.append(floating_ip)
+        except:
```

Comment:
What exactly do you want to catch here?

Reply:
Catching the refusal of IP allocation. 
I need to allocate them all to catch the exception because I did not see it in the documentation of libcloud